sunpy/coordinates/frames.py

```diff
@@ -470,6 +470,26 @@ class Helioprojective(SunPyBaseCoordinateFrame):
     rsun = QuantityAttribute(default=_RSUN, unit=u.km)
     observer = ObserverCoordinateAttribute(HeliographicStonyhurst)
 
+    @property
+    def angular_radius(self):
+        """
+        Angular radius of the Sun as seen by the observer.
+
+        The ``rsun`` frame attribute is the radius of the Sun in length units.
+        The tangent vector from the observer to the edge of the Sun forms a
+        right-angle triangle with the radius of the Sun as the far side and the
+        Sun-observer distance as the hypotenuse.  Thus, the sine of the angular
+        radius of the Sun is ratio of these two distances.
+        """
+        from sunpy.coordinates.sun import _angular_radius  # avoiding a circular import
+
+        if not isinstance(self.observer, HeliographicStonyhurst):
+            if self.observer is None:
+                raise ValueError("The observer must be defined, not `None`.")
+            raise ValueError("The observer must be fully defined by specifying `obstime`.")
```

Comment:
Can you add a test that raises each of these warnings?

Reply:
Sure on it !

Reply:
hey, can you tell me something that is an instance of HeliographicStonyhurst?

Reply:
Don't know myself but https://docs.sunpy.org/en/stable/api/sunpy.coordinates.HeliographicStonyhurst.html might be useful.

Reply:
The two error cases to test are:

- `Helioprojective(observer=None).angular_radius`
- `Helioprojective(observer='earth', obstime=None).angular_radius`